src/WidgetFactory.php

```diff
@@ -55,14 +80,74 @@ public static function initialize(
      * @psalm-suppress MixedInferredReturnType
      * @psalm-suppress MixedReturnStatement
      */
-    public static function createWidget(array $config): Widget
+    public static function createWidget(array $config, ?string $theme = null): Widget
     {
         if (self::$factory === null) {
             throw new WidgetFactoryInitializationException(
                 'Widget factory should be initialized with WidgetFactory::initialize() call.',
             );
         }
 
+        $theme ??= self::$defaultTheme;
+
+        if ($theme !== null && isset(self::$themes[$theme])) {
+            if (
+                is_string($config[ArrayDefinition::CLASS_NAME])
+                && isset(
+                    $config[ArrayDefinition::CLASS_NAME],
+                    self::$themes[$theme][$config[ArrayDefinition::CLASS_NAME]]
+                )
+            ) {
+                $config = ArrayDefinitionHelper::merge(
+                    self::$themes[$theme][$config[ArrayDefinition::CLASS_NAME]],
+                    $config
+                );
+            }
```

Comment:
```suggestion
            $className = $config[ArrayDefinition::CLASS_NAME] ?? null;
            if ($className !== null && isset(self::$themes[$theme][$className])) {
                $config = ArrayDefinitionHelper::merge(
                    self::$themes[$theme][$className],
                    $config
                );
            }
```

Reply:
Replace `$className !== null` to `is_string($className)` and apply it.